### PR TITLE
Initial do-not-submit script and action.yml

### PR DESCRIPTION
**Pull request by inFocus7**
idea is to run this like

```
   - name: Check for unimplemented work
      uses: solo-io/do-not-submit@v1
      with
          keyword: DO_NOT_SUBMIT
          allow_list: *.go,go.mod,*.proto
```

### CODE REVIEW

**Comment by inFocus7** [comment]
maybe `check_list` instead of `allow_list` would be more understandable :hmm:

File: do-not-submit.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+KEYWORD=$1
+CHECK_LIST=$2
+IGNORE_LIST=$3
+# Gets the rest of the arguments - which should be the list of files - as an array.
+FILES=("${@:4}")
+
+# TODO: What would be a good separator? `,`, `|`, or google's way and use `\n` here? (https://github.com/google-github-actions/get-secretmanager-secrets#inputs)
+IFS='|' read -ra check_list_array <<< "$CHECK_LIST"
+IFS='|' read -ra ignore_list_array <<< "$IGNORE_LIST"
```
Comment (inFocus7) [inline do-not-submit.sh]: TODO: Need to change the separator to `,`

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Fetch changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v37
+    - name: Check files for ${{ inputs.keyword }}
+      id: check-files
+      shell: bash
+      # TODO: Does `all_modified_files` include new, copied, and type_changed files?
```
Comment (jbohanon) [inline action.yml]: Does this help clarify? https://github.com/marketplace/actions/changed-files?version=v37#outputs

Comment (jbohanon) [reply action.yml]: FYI looks like all_modified_files was removed between v35 and v37

Comment (inFocus7) [reply action.yml]: Lol yeah. Your latest changes from `all_modified_files` to `all_changed_files` is what clicked for me to update this 🍾

**Pull request merged**